### PR TITLE
ES Modules

### PR DESCRIPTION
**Pull request by avoc-ado**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,15 +1,18 @@
 {
-  "name": "@tensor-hq/tensor-common",
-  "version": "0.19.9",
+  "name": "@tensor-hq/tensor-common-alpha",
```
Comment (richardwu) [inline package.json:2]: Revert back to common

File: package.json [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
+  "version": "0.19.0-alpha.15",
```
Comment (richardwu) [inline package.json]: Bump this to 1.0.0

File: src/lambdas/trait_bid_cosigner_serde.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import BN from 'bn.js';
 import { PublicKey, TransactionInstruction } from '@solana/web3.js';
 import { Creator, UseMethod } from '@metaplex-foundation/mpl-token-metadata';
-import { MetadataArgs } from '@metaplex-foundation/mpl-bubblegum';
+import type { MetadataArgs } from '@metaplex-foundation/mpl-bubblegum';
```
Comment (richardwu) [inline src/lambdas/trait_bid_cosigner_serde.ts:4]: 👌

File: src/metaplex/bubblegum.ts [hunk 4]
```diff
@@ -1,11 +0,0 @@
...
-  MetadataArgs,
-  metadataArgsBeet,
-} from '@metaplex-foundation/mpl-bubblegum';
-import { keccak_256 } from 'js-sha3';
-
-/** Version from metaplex but without seller fee basis points */
-export function computeMetadataArgsHash(metadata: MetadataArgs): Buffer {
-  const [serializedMetadata] = metadataArgsBeet.serialize(metadata);
-  return Buffer.from(keccak_256.digest(serializedMetadata));
-}
```
Comment (richardwu) [inline src/metaplex/bubblegum.ts:11]: Hmmm aren't we using this in tensor-infra?

File: src/metaplex/pdas/token_rules.ts [hunk 5]
```diff
@@ -1,22 +1,22 @@
 import { PublicKey } from '@solana/web3.js';
-export { PROGRAM_ID as TMETA_PROG_ID } from '@metaplex-foundation/mpl-token-metadata';
-export { PROGRAM_ID as AUTH_PROG_ID } from '@metaplex-foundation/mpl-token-auth-rules';
-import { PROGRAM_ID as TMETA_PROG_ID } from '@metaplex-foundation/mpl-token-metadata';
-import {
-  PREFIX,
-  PROGRAM_ID as AUTH_PROG_ID,
-} from '@metaplex-foundation/mpl-token-auth-rules';
+import { TOKEN_METADATA_PROGRAM_ID } from 'src/programs/auction_house/shared';
```
Comment (richardwu) [inline src/metaplex/pdas/token_rules.ts:2]: Should do a relative import here?

File: src/metaplex/pdas/token_rules.ts [hunk 6]
```diff
@@ -1,22 +1,22 @@
...
+
+export const AUTH_PROG_ID = new PublicKey(
+  'auth9SigNpDKz4sJJ1DfCTuZrZNSAgh9sFD3rboVmgg',
+);
+
+const PREFIX = 'rule_set';
```
Comment (richardwu) [inline src/metaplex/pdas/token_rules.ts:8]: Move this to where it's called

File: src/programs/hadeswap/pda.ts [hunk 7]
```diff
@@ -1,14 +1,10 @@
...
-import { hadeswap } from 'hadeswap-sdk';
+import { HADESWAP_ADDR, HADESWAP_FEE_PREFIX } from './constants';
 
 export const findHSwapFeeVaultPda = (pair: PublicKey) => {
   return PublicKey.findProgramAddressSync(
-    [
-      Buffer.from(utils.bytes.utf8.encode(hadeswap.constants.FEE_PREFIX)),
-      pair.toBytes(),
-    ],
+    [Buffer.from(utils.bytes.utf8.encode(HADESWAP_FEE_PREFIX)), pair.toBytes()],
```
Comment (richardwu) [inline src/programs/hadeswap/pda.ts:7]: 🔥

File: src/programs/hadeswap/price.ts [hunk 8]
```diff
@@ -1,9 +1,59 @@
...
+    const currentDelta = delta + 1 - counterUpdated;
+    const diffAmount = (counterUpdated * nftTokensBalance) / currentDelta;
+    const newNftTokensBalance = nftTokensBalance + diffAmount;
+
+    return orderType === HadeswapOrderType.Buy
+      ? newNftTokensBalance / (currentDelta - 1)
+      : newNftTokensBalance / (currentDelta + 1);
+  }
+  return 0;
+};
```
Comment (richardwu) [inline src/programs/hadeswap/price.ts:56]: You pulled this from the bundled code here right? https://www.npmjs.com/package/hadeswap-sdk?activeTab=code

File: src/solana_contrib/anchor.ts [hunk 9]
```diff
@@ -5,22 +5,14 @@ import {
...
-  TransactionResponse,
-} from '@solana/web3.js';
+import type { InstructionDisplay } from '@coral-xyz/anchor/dist/cjs/coder/borsh/instruction';
+import type { AllAccountsMap } from '@coral-xyz/anchor/dist/cjs/program/namespace/types';
+import { AccountInfo, PublicKey, TransactionResponse } from '@solana/web3.js';
 import bs58 from 'bs58';
 import { ExtractedIx, extractAllIxs } from './transaction';
 
-export const dummyWallet = () => new Wallet(Keypair.generate());
-
```
Comment (richardwu) [inline src/solana_contrib/anchor.ts:23]: So I guess we moved this to tensor-infra?

**Review by richardwu — CHANGES_REQUESTED** [review]

File: src/metaplex/pdas/token_metadata.ts [hunk 10]
```diff
@@ -1,11 +1,11 @@
-import { PROGRAM_ID } from '@metaplex-foundation/mpl-token-metadata';
 import { PublicKey } from '@solana/web3.js';
 import BN from 'bn.js';
+import { TOKEN_METADATA_PROGRAM_ID } from 'src/programs';
```
Comment (richardwu) [inline src/metaplex/pdas/token_metadata.ts:3]: Should do relative imports right? Otherwise i think it breaks once we transpile

**Review by richardwu — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (avoc-ado) [reply package.json]: 1.0.0-rc.1 till its tested

File: src/metaplex/bubblegum.ts (hunk 4 above)
Comment (avoc-ado) [reply src/metaplex/bubblegum.ts:11]: i'll move it in infra if so

File: src/programs/hadeswap/price.ts (hunk 8 above)
Comment (avoc-ado) [reply src/programs/hadeswap/price.ts:56]: from here, updated the comment
 https://github.com/hadeswap-solana/hadeswap-sdk-public/blob/66ab430a7a11bd014804d48a7f5d2e7fd906f9ac/src/hadeswap-core/helpers.ts#L86

File: src/solana_contrib/anchor.ts (hunk 9 above)
Comment (avoc-ado) [reply src/solana_contrib/anchor.ts:23]: yup, the frontend complains that anchor doesn't export Wallet. It's exported in a funny way

File: src/solana_contrib/utils.ts [hunk 11]
```diff
@@ -10,16 +10,23 @@ export const validPublicKey = (input: string) => {
...
 // based on https://docs.solana.com/developing/programming-model/accounts#:~:text=The%20current%20maximum%20size%20of,per%20account%20and%20per%20instruction.
-export const getAccountRentSync = (dataSize: number) =>
+export const getRentSync = (dataSize: number) =>
   Math.trunc(19.055441478439427 * (128 + dataSize) * 365.25);
 
+export const getLamports = async (
+  conn: Connection,
+  acct: PublicKey,
+): Promise<number | undefined> => {
+  return (await conn.getAccountInfo(acct))?.lamports;
```
Comment (richardwu) [inline src/solana_contrib/utils.ts:27]: Isn't there a `getBalance` fn?

**Review by richardwu — APPROVED** [review]

File: src/solana_contrib/utils.ts (hunk 11 above)
Comment (avoc-ado) [reply src/solana_contrib/utils.ts:27]: it returns a string 🤮

**Pull request merged**